extensions/wikia/Flags/controllers/FlagsApiController.class.php

```diff
@@ -132,22 +132,19 @@ public function getFlagsForPageForEdit() {
 	 * ]
 	 */
 	public function addFlagsToPage() {
-		try {
-			$this->processRequest();
+		$this->processRequest();
+		$this->validatePageId();
+		$this->checkUserEditPermissions( $this->params['page_id'], $this->params['wiki_id'] );
 
-			$flagModel = new Flag();
-			$modelResponse = $flagModel->addFlagsToPage( $this->params );
+		$flagModel = new Flag();
+		$modelResponse = $flagModel->addFlagsToPage( $this->params );
 
-			$this->getCache()->purgeFlagsForPage( $this->params['page_id'] );
-			$this->purgeFlaggedPages( $this->params['flags'] );
-			$this->purgeFlagInsights( $this->params['flags'] );
+		$this->getCache()->purgeFlagsForPage( $this->params['page_id'] );
+		$this->purgeFlaggedPages( $this->params['flags'] );
```

Comment:
I can see a couple of calls to $this->params['flags'], maybe it's result could be cached in a local var?